### ISSUE TITLE
CashFlow cannot be negative

### ISSUE THREAD

**Issue by damian-t** [ta4j/ta4j#332]
The `CashFlow` class seems to be wrong for short trades (two orders: first `OrderType.SELL`, followed by `OrderType.BUY`) as it will never be negative. In reality for short trades, unlimited losses are possible and profits are capped at 100%.

It can be seen e.g. in `CashFlowTest.cashFlowWithSellAndBuyOrders()`:
The last trade is:
- Selling the asset for 3$ (at index 5)
- Buying the asset for 20$ (at index 6)
The reported 1-period cashflow is `3/20 = 0.15`, standing for a 1-period return of -85%. However it should be `-(20/3-1)+1 = -4.66`, standing for a 1-period return of -566%.

Another way of seeing this, is imagining selling 20 assets for 3$ (gaining a total of 60$) and buying 20 assets for 20$ (spending the 60$ earned earlier plus 340$) to settle the trade. Since I had to pay more than 60$ out of my own pocket, I 'lost' more than 100%.

Edit: On second thought, I think the whole "multiplicative" setup of the `CashFlow` (see `CashFlow.calculate() {... values.add(values.get(entryIndex).multipliedBy(ratio));}`) class won't work for short trades as one could not recover from a total loss.

**Comment by damian-t** [comment]
I brooded a bit more about this. My current view is the following: For proper modelling of a strategy involving short selling, we would have to model the margin account (incl. margin call) which would be quite a substantial extension.

One option that could be implemented relatively easily, is the following:
We assume there is no margin account but as soon as the losses of a short sell are realized (the short trade is closed), the strategy ends - the cash flow remains at that level.

Let's look at `ChashFlowTest.cashFlowShortSell()`: 
- The first trade (long): increases the initial capital from 1 to 4
- The second trade (short): decreases the capital from 4 to -8. Then the trade is closed and the loss realized. 
- Since we don't model the margin account, we have no 'additional' capital available to enter into the third (long) trade. Therefore, the cashflow balance remains constant.
- The cashflow sequence is then: `1, 2, 4, 0, -8, -8`

**Comment by team172011** [comment]
@damian-t did you already fixed this issue in scope of #362?

**Comment by milczarekIT** [comment]
Hi @team172011 , @damian-t .
Yes described by @damian-t issue was fixed in #362.
I added some missing tests for Cashflow class in https://github.com/ta4j/ta4j/pull/489 
Also with described by @damian-t scenario:

```
    @Test
    public void cashFlowShortSellBigLossWithNegativeCashFlow() {
        BarSeries sampleBarSeries = new MockBarSeries(numFunction, 3, 20);
        TradingRecord tradingRecord = new BaseTradingRecord(Order.sellAt(0, sampleBarSeries),
                Order.buyAt(1, sampleBarSeries));

        CashFlow cashFlow = new CashFlow(sampleBarSeries, tradingRecord);

        assertNumEquals(1, cashFlow.getValue(0));
        assertNumEquals(-4.6667, cashFlow.getValue(1));
    }
```
